File: client-examples/csharp/SmartCompress.cs
using System;
using System.Net.Http;
using System.IO;
using System.Threading.Tasks;
using System.Collections.Generic;

namespace SmartCompress
{
    public class Client
    {
        private readonly HttpClient _httpClient;
        private readonly string _apiKey;

        public Client(string apiKey)
        {
            _apiKey = apiKey;
            _httpClient = new HttpClient();
            _httpClient.DefaultRequestHeaders.Add("X-API-Key", _apiKey);
        }

        public async Task FromFileAsync(string inputPath, string outputPath, Dictionary<string, object> options = null)
        {
            using var form = new MultipartFormDataContent();
            using var fileStream = File.OpenRead(inputPath);
            form.Add(new StreamContent(fileStream), "image", Path.GetFileName(inputPath));

            if (options != null)
            {
                foreach (var kvp in options)
                {
                    form.Add(new StringContent(kvp.Value.ToString()), kvp.Key);
                }
            }

            var response = await _httpClient.PostAsync("https://shrinkix.com/api/compress", form);

            if (!response.IsSuccessStatusCode)
            {
                var error = await response.Content.ReadAsStringAsync();
                throw new Exception($"Compression failed: {response.StatusCode} - {error}");
            }

            using var outputStream = await response.Content.ReadAsStreamAsync();
            using var outputFile = File.Create(outputPath);
            await outputStream.CopyToAsync(outputFile);
        }
    }
}

File: client-examples/ruby/lib/smartcompress.rb
require 'net/http'
require 'uri'
require 'json'
require 'net/http/post/multipart'

class SmartCompress
  def initialize(api_key)
    @api_key = api_key
    @url = URI.parse('https://api.shrinkix.com/compress')
  end

  def from_file(input_path, output_path, options = {})
    File.open(input_path) do |file|
      params = {
        "image" => UploadIO.new(file, "image/#{File.extname(input_path).delete('.')}", File.basename(input_path))
      }
      
      params["quality"] = options[:quality].to_s if options[:quality]
      params["width"] = options[:width].to_s if options[:width]
      params["height"] = options[:height].to_s if options[:height]
      
      req = Net::HTTP::Post::Multipart.new(@url.path, params)
      req['X-API-Key'] = @api_key

      res = Net::HTTP.start(@url.host, @url.port, use_ssl: true) do |http|
        http.request(req)
      end

      if res.code == '200'
        File.binwrite(output_path, res.body)
        return true
      else
        raise "Compression failed: #{res.body}"
      end
    end
  end
end
